### PR TITLE
Load conversations

### PR DESCRIPTION
**Pull request by amaury901130**
#### Description:

* Load conversations and messages connection.

---

### CODE REVIEW

File: RSDemoProject/Networking/Services/ConversationResource.swift [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+//  Created by Amaury Ricardo on 6/10/20.
+//  Copyright © 2020 TopTier labs. All rights reserved.
+//
+
+import Foundation
+import Moya
+
+enum ConversationResource: TargetType {
+  
+  case getConversations
```
Comment (germanStabile) [inline RSDemoProject/Networking/Services/ConversationResource.swift:14]: conversations should also be paginated right?

Comment (amaury901130) [reply RSDemoProject/Networking/Services/ConversationResource.swift:14]: No, the api documentation has just pagination for the messages.

File: RSDemoProject/Home/ViewModels/MainViewModel.swift [hunk 2]
```diff
@@ -27,13 +28,18 @@ class MainViewModel {
   private var page = 1
   var defaultMapRadius = 200
   var userTargets: [Target] = []
+  var conversations: [Conversation]?
```
Comment (glm4) [inline RSDemoProject/Home/ViewModels/MainViewModel.swift:31]: optional not needed, an empty array is preferred

File: RSDemoProject/Home/Views/MainViewController.swift [hunk 3]
```diff
@@ -12,6 +12,7 @@ import MapKit
 class MainViewController: UIViewController {
   
   let mainTitleSpacing = 1.95
+  let conversationCount: CGFloat = 6
```
Comment (glm4) [inline RSDemoProject/Home/Views/MainViewController.swift:15]: why do you need the default value?

File: RSDemoProject/Networking/Services/ConversationResource.swift [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  case getMessages(conversationId: Int, page: Int = 1)
+  
+  var path: String {
+    let conversatioBaseUrl = "match_conversations"
```
Comment (glm4) [inline RSDemoProject/Networking/Services/ConversationResource.swift:18]: typo in constant name

File: RSDemoProject/Networking/Services/ConversationService.swift [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  
+  func getConversations(
+    _ success: @escaping (_ response: [Conversation]) -> Void,
+    failure: @escaping (_ error: Error) -> Void
+  ) {
+    request(
+      for: .getConversations,
+      onSuccess: { (result: [Conversation], _) -> Void in
+        success(result)
+    },
```
Comment (glm4) [inline RSDemoProject/Networking/Services/ConversationService.swift:23]: closing braces should be aligned with block alias.

File: RSDemoProject/Networking/Services/ConversationService.swift [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    page: Int,
+    success: @escaping (_ response: [Message]) -> Void,
+    failure: @escaping (_ error: Error) -> Void
+  ) {
+    request(
+      for: .getConversations,
+      at: "messages",
+      onSuccess: { (result: [Message], _) -> Void in
+        success(result)
+    },
```
Comment (glm4) [inline RSDemoProject/Networking/Services/ConversationService.swift:40]: same here

File: RSDemoProject/Home/Views/MainViewController.swift (hunk 3 above)
Comment (amaury901130) [reply RSDemoProject/Home/Views/MainViewController.swift:15]: This is the radius

**Comment by amaury901130** [comment]
> I think a different order of implementation would be better. I don't believe you will need the list of conversations in the main scree/view model. You will need to refactor in next PRs when implementing the conversations screen after all.

@glm4 I need to load it here because I need to count the unread conversations

**Comment by glm4** [comment]
> > I think a different order of implementation would be better. I don't believe you will need the list of conversations in the main screen/view model. You will need to refactor in the next PRs when implementing the conversations screen after all.
> 
> @glm4 I need to load it here because I need to count the unread conversations

Considering the API is missing a lot of features and that this should not be the best way to get that number, you still don't need the list of conversation objects, just to store the unread count.

**Comment by amaury901130** [comment]
> > > I think a different order of implementation would be better. I don't believe you will need the list of conversations in the main screen/view model. You will need to refactor in the next PRs when implementing the conversations screen after all.
> > 
> > 
> > @glm4 I need to load it here because I need to count the unread conversations
> 
> Considering the API is missing a lot of features and that this should not be the best way to get that number, you still don't need the list of conversation objects, just to store the unread count.

Agree, but there is no other way, What I want to do next, is storage that conversations in a local DB, and then when the user opens the conversation list he already has the conversations loaded in memory.

**Comment by glm4** [comment]
> > > > I think a different order of implementation would be better. I don't believe you will need the list of conversations in the main screen/view model. You will need to refactor in the next PRs when implementing the conversations screen after all.
> > > 
> > > 
> > > @glm4 I need to load it here because I need to count the unread conversations
> > 
> > 
> > Considering the API is missing a lot of features and that this should not be the best way to get that number, you still don't need the list of conversation objects, just to store the unread count.
> 
> Agree, but there is no other way, What I want to do next, is storage that conversations in a local DB, and then when the user opens the conversation list he already has the conversations loaded in memory.

That's ok. What you are saying is the scope of next features and PRs. For this one specifically, you don't need all conversations in memory. You don't add code you 'might use in the future'.

File: RSDemoProject/Home/Views/MainViewController.swift [hunk 7]
```diff
@@ -120,6 +126,16 @@ class MainViewController: UIViewController {
...
   }
+  
+  func showUnreadMessages() {
+    guard viewModel.unreadMessages > 0 else {
+      unreadMessagesLabel.isHidden = true
+      return
+    }
+
+    unreadMessagesLabel.text = String(viewModel.unreadMessages)
+    unreadMessagesLabel.isHidden = false
```
Comment (glm4) [inline RSDemoProject/Home/Views/MainViewController.swift:137]:
```
unreadMessagesLabel.isHidden = viewModel.unreadMessages <= 0
unreadMessagesLabel.text = "\(viewModel.unreadMessages)"
```

File: RSDemoProject/Home/Views/MainViewController.swift [hunk 8]
```diff
@@ -120,6 +126,16 @@ class MainViewController: UIViewController {
       )
     }
   }
+  
+  func showUnreadMessages() {
```
Comment (glm4) [inline RSDemoProject/Home/Views/MainViewController.swift:130]: this method should be named toggle or similar.

**Pull request merged**